nomad/secure_variables_endpoint.go

```diff
@@ -367,3 +368,95 @@ func (sv *SecureVariables) decrypt(v *structs.SecureVariable) error {
 	v.EncryptedData = nil
 	return nil
 }
+
+// handleMixedAuthEndpoint is a helper to handle auth on RPC endpoints that can
+// either be called by external clients or by workload identity
+func (sv *SecureVariables) handleMixedAuthEndpoint(args structs.QueryOptions, cap, pathOrPrefix string) error {
+
+	// Perform the initial token resolution.
+	aclObj, err := sv.srv.ResolveToken(args.AuthToken)
+	if err == nil {
+		// Perform our ACL validation. If the object is nil, this means ACLs
+		// are not enabled, otherwise trigger the allowed namespace function.
+		if aclObj != nil {
+			if !aclObj.AllowNsOp(args.RequestNamespace(), cap) {
+				return structs.ErrPermissionDenied
+			}
+		}
+		return nil
+	}
+	if helper.IsUUID(args.AuthToken) {
+		// early return for ErrNotFound or other errors if it's formed
+		// like an ACLToken.SecretID
+		return err
+	}
+
+	// Attempt to verify the token as a JWT with a workload
+	// identity claim
+	claim, err := sv.srv.ResolveClaim(args.AuthToken)
+	if err != nil {
+		return structs.ErrPermissionDenied
+	}
+
+	alloc, err := sv.authValidateAllocationIdentity(claim.AllocationID, args.RequestNamespace())
+	if err != nil {
+		metrics.IncrCounter([]string{
+			"nomad", "secure_variables", "invalid_allocation_identity"}, 1)
+		sv.logger.Trace("allocation identity was not valid", "error", err)
+		return structs.ErrPermissionDenied
+	}
+
+	err = sv.authValidatePrefix(alloc, claim.TaskName, pathOrPrefix)
+	if err != nil {
+		sv.logger.Trace("allocation identity did not have permission for path")
```

Comment:
```suggestion
		sv.logger.Trace("allocation identity did not have permission for path", "error", err)
```

Right?

Reply:
The error is _always_ `ErrPermissionDenied`, which is a little redundant. But given this is a trace log, it can't hurt to make the logger a little future-proof for new errors.